src/sweeper/workspace_info.py

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env python
+# * coding: utf8 *
+'''
+workspace_info.py
+A module that gets information about the workspace, including a list of feature classes.
+'''
+import arcpy
+import os
+
+
+#: A function to generate and return a list of workspace feature classes.
+def get_featureclasses(workspace_path):
+    '''
+    workspace_path: full path to the feature workspace.
+    '''
+    arcpy.env.workspace = workspace_path
+    fc_list = []
+
+    datasets = arcpy.ListDatasets(feature_type='feature')
+    datasets = [''] + datasets if datasets is not None else []
+
+    #: generate a list of feature classes showing their full path (Example: Directory/Workspace/Dataset/TableName)
+    for ds in datasets:
+        for fc in arcpy.ListFeatureClasses(feature_dataset=ds):
+            fc_list.append(fc)
```

Comment:
does this get stand alone feature classes if there are no datasets? It looks a little confusing to me with the `['']` and the else being `[]`. I'm curious if there will always be an empty ds to get the stand alones or not.

Reply:
it was totally confusing to me to, but yes it does get standalone if they are not in datasets.   i copied it from the bottom of this page. https://pro.arcgis.com/en/pro-app/arcpy/functions/listfeatureclasses.htm

Reply:
ok works for me. https://pro.arcgis.com/en/pro-app/arcpy/data-access/walk.htm is another alternative to be aware of.